numpy/meson.build

```diff
@@ -143,12 +145,45 @@ foreach subdir: pure_subdirs
   install_subdir(subdir, install_dir: np_dir)
 endforeach
 
-custom_target('__config__.py',
-  output: '__config__.py',
+cc = meson.get_compiler('c')
+cpp = meson.get_compiler('cpp')
+compilers = {'C': cc, 'CPP': cpp}
+
+conf_data = configuration_data()
+
+# Set compiler information
+foreach name, compiler : compilers
+  conf_data.set(name + '_COMP', compiler.get_id())
+  conf_data.set(name + '_COMP_LINKER_ID', compiler.get_linker_id())
+  conf_data.set(name + '_COMP_VERSION', compiler.version())
+  conf_data.set(name + '_COMP_CMD_ARRAY', compiler.cmd_array())
+endforeach
+
+# Host CPU and system information
+conf_data.set('HOST_CPU', host_machine.cpu())
+conf_data.set('HOST_CPU_FAMILY', host_machine.cpu_family())
+conf_data.set('HOST_CPU_ENDIAN', host_machine.endian())
+conf_data.set('HOST_CPU_SYSTEM', host_machine.system())
+
+# Python information
```

Comment:
We could perhaps add some more info here using https://mesonbuild.com/Python-module.html

Reply:
Added on [a4006f2](https://github.com/numpy/numpy/pull/22769/commits/a4006f2af87295a44808fb1a927f7561909688d9)